packages/enzyme/src/ShallowWrapper.js

```diff
@@ -261,6 +310,25 @@ class ShallowWrapper {
         throw new TypeError('ShallowWrapper can only wrap valid elements');
       }
 
+      if (typeof options.wrappingComponent === 'function') {
+        if (typeof adapter.wrapWithWrappingComponent !== 'function') {
+          throw new TypeError('your adapter does not support `wrappingComponent`. Try upgrading it!');
+        }
+        const { node: wrappedNode, RootFinder } = adapter.wrapWithWrappingComponent(nodes, options);
```

Comment:
Previously the logic in `adapter.wrapWithWrappingComponent` was in enzyme. This method is all about element creation, so I think it did make more sense for it to live in the adapter.